### PR TITLE
Shoot command 3

### PR DESCRIPTION
**Pull request by Arnav814**
## Overview

This PR will add a new command that shoots using pre-calibrated settings from a known position.
Also fixes some commands that didn't extend `Command`.

## Tests Ran

It builds

### CODE REVIEW

File: src/main/java/frc/robot/commands/gpm/IndexerFeed.java [hunk 1]
```diff
@@ -1,31 +1,36 @@
...
     }
 
+	@Override
     public boolean isFinished(){
 
         return !m_index.hasNote();
         
     }
 
+	@Override
```
Comment (Kyle-Eldridge) [inline src/main/java/frc/robot/commands/gpm/IndexerFeed.java:33]: These don't need to be indented. You don't have to change them back for this PR, but in the future you could leave them unindented.

File: src/main/java/frc/robot/commands/gpm/ShootKnownPos.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+public class ShootKnownPos extends SequentialCommandGroup {
+	
+	/**
+	* A preset shot that can be taken by the {@link frc.robot.commands.gpm.ShootKnownPos} command
+	*/
+	public enum ShotPosition {
+		// TODO: add actual values
+		SUBWOOFER_TOP(0.0, 0.0),
+		SUBWOOFER_MIDDLE(0.0, 0.0),
+		SUBWOOFER_BOTTOM(0.0, 0.0);
```
Comment (Kyle-Eldridge) [inline src/main/java/frc/robot/commands/gpm/ShootKnownPos.java:20]: For the shooter, these should all be the same. You can probably leave them for now until we figure out what that position is in the next PR.

File: src/main/java/frc/robot/commands/gpm/ShootKnownPos.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+		addRequirements(shooter, arm, storageIndex);
+
+		addCommands(
+			new ParallelCommandGroup(
+				new ArmToPos(arm, shot.getArmAngle()),
+				new SetShooterSpeed(shooter, shot.getShooterSpeed())),
+			new IndexerFeed(storageIndex),
+			new ParallelCommandGroup(
+				new ArmToPos(arm, ArmConstants.stowedSetpoint),
+				new PrepareShooter(shooter, 0)));
```
Comment (Kyle-Eldridge) [inline src/main/java/frc/robot/commands/gpm/ShootKnownPos.java:57]: The last parallel command group should have instant commands so we can start another command while the arm and shooter are going back to their setpoints (we don't want to wait for the shooter to slow down all the way back to 0 before another command can use the arm and indexer).

**Review by Kyle-Eldridge — APPROVED** [review]

File: src/main/java/frc/robot/commands/gpm/ShootKnownPos.java
Comment (Ash1R) [inline src/main/java/frc/robot/commands/gpm/ShootKnownPos.java:1]: Would we use prepare shooter here, or set shooter? I could be wrong, but preparing the shooter implies that its done before shooting?
To my knowledge, we're going to have a button to get the shooter wheels to a certain speed (that would use prepare command), and a separate button for the shot itself.

Comment (Arnav814) [reply src/main/java/frc/robot/commands/gpm/ShootKnownPos.java:1]: The only difference between the two is that prepare shooter exits immediately, whereas set shooter waits for the shooter to spin up. Better names would probably help.

Comment (Ash1R) [reply src/main/java/frc/robot/commands/gpm/ShootKnownPos.java:1]: I see, that's an easy fix :) Does that also helpswith one of kyle's comments about waiting for shooter to spin down?

Comment (Arnav814) [reply src/main/java/frc/robot/commands/gpm/ShootKnownPos.java:1]: It does, but we still need to make the arm command exit instantly.

File: src/main/java/frc/robot/commands/gpm/ShootKnownPos.java (hunk 2 above)
Comment (Ash1R) [reply src/main/java/frc/robot/commands/gpm/ShootKnownPos.java:21]: small chance they might be different as well. Leave for now, fix later if neccessary

**Review by Ash1R — APPROVED** [review]

**Pull request merged**